src/app/shared-services/db/sqlite.service.ts

```diff
@@ -112,29 +112,57 @@ export class SqliteService {
       },
     ]
 
-    // insert or update new points query
-    const numPoints = t.coordinates.length
-    if (numPoints) {
-      // construct query & values array
-      const placeholders = []
-      const values = []
-      for (let i = 0; i < numPoints; i++) {
-        const time = t.timestamps[i]
-        const [lat, lon] = t.coordinates[i]
-        const accuracy = t.accuracy[i]
-        placeholders.push(`(?,?,?,?,?)`)
-        values.push(t.id, time, lat, lon, accuracy)
-      }
-      const placeholderString = placeholders.join(', ')
-      const statement = `INSERT OR REPLACE INTO points VALUES ${placeholderString}`
-      set.push({ statement, values: values.map(normalize) })
-    }
-
     const {
       changes: { changes },
       message,
     } = await this.db.executeSet({ set })
     if (changes === -1) throw new Error(`couldnt insert trajectory: ${message}`)
+
+    await this.upsertPointsForTrajectory(t)
+  }
+
+  async upsertPointsForTrajectory(t: Trajectory) {
+    // insert or update new points query
+    const numPoints = t.coordinates.length
+    if (numPoints) {
+      // construct query & values array in chunks to prevent to many sql-statements at once
+      // since the length limit of a query 'SQLITE_MAX_SQL_LENGTH' defaults to 1 000 000
+      const chunkSize = 1000
```

Comment:
This chunk size seems to work on my tested platforms.

But looking at the sqlite docs, our limit might actually be 999. See `Maximum Number Of Host Parameters In A Single SQL Statement` in https://sqlite.org/limits.html

Reply:
I'll look into it - shouldn't make a significant difference to reduce this number a little bit.

Reply:
Just checked: the plugin we are using uses SQLite v3.33.0 and for that the `Maximum Number Of Host Parameters In A Single SQL Statement` defaults to 32766.

> [...] defaults to 999 for SQLite versions prior to 3.32.0 (2020-05-22) or 32766 for SQLite versions after 3.32.0.